interface/main/finder/dynamic_finder_ajax.php

```diff
@@ -63,26 +120,26 @@
 // Global filtering.
 //
 $where = '';
+$sSearch = add_escape_custom(trim($_GET['sSearch']));
+$cnt_columns = count($aColumns);
+
 if (isset($_GET['sSearch']) && $_GET['sSearch'] !== "") {
-    $sSearch = add_escape_custom(trim($_GET['sSearch']));
-    foreach ($aColumns as $colname) {
-        $where .= $where ? "OR " : "WHERE ( ";
+    foreach ($aColumns as $index_col => $colname) {
+        $where .= $where ? ($cnt_columns != $index_col ? "OR " : "" ) : "WHERE ( ";
+
```

Comment:
ok, I think we are getting close to a nice generic mechanism to support both dates and lists:
1. See if the $colname is set to any row in layout_options table where form_id = DEM . Then look at the data_type for that row:
 -If it is 4, then that is a date and you will use your date thingy for that $colname. You are then done for these.
 -If it is 1, then you have a list of which the name of the list is stored in list_id that maps to the list_options table. Then what you will do is build a php array of the list in list_options with following elements:
1. list_id
2. title
3. xl_list_label(title)
You can then use this to map your language specific search string from the xl_list_label(title) back to list_id (and this mapped item(s) can be used to do the actual sql query)

I hope this makes sense.

Reply:
hi @yurige-matrix ,
Forgot to tag you in above post that I just posted. thanks, -brady

Reply:
Hi @bradymiller. I see that @sjpadgett  has already done everything for this task.

Reply:
hi @yurige-matrix ,
Nice. Didn't realize @sjpadgett code would cover this. Let me know if I can close this PR.